cmd/soroban-rpc/internal/methods/json.go

```diff
@@ -58,3 +58,20 @@ func transactionToJSON(tx db.Transaction) (
 
 	return result, envelope, resultMeta, nil
 }
+
+func ledgerToJSON(meta xdr.LedgerCloseMeta, header xdr.LedgerHeaderHistoryEntry) ([]byte, []byte, error) {
+	var err error
+	var closeMetaJSON, headerJSON []byte
+
+	closeMetaJSON, err = xdr2json.ConvertInterface(meta)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	headerJSON, err = xdr2json.ConvertInterface(header)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return closeMetaJSON, headerJSON, nil
+}
```

Comment:
A few comments here:

1. These can be quite large, it might be worth passing a pointer to avoid the stack copy for every ledger?
2. The header is a subset of the meta, so you shouldn't need to pass that in. 
3. If you don't pass it in, you can either (a) convert it separately like you're doing here or (b) actually extract it from the `closeMetaJSON` and return that, instead. The latter is probably more efficient but it's error-prone since it relies on the JSON structure not changing.